symengine-hs.cabal

```diff
@@ -0,0 +1,46 @@
+name:                symengine-hs
+version:             0.1.1.0
+synopsis:            SymEngine symbolic mathematics engine for Haskell
+description:         Please see README.md
+homepage:            http://github.com/bollu/symengine.hs#readme
+license:             MIT
+license-file:        LICENSE
+author:              Siddharth Bhat
+maintainer:          siddu.druid@gmail.com
+copyright:           2016 Siddharth Bhat
+category:            FFI, Math, Symbolic Computation
+build-type:          Simple
+-- extra-source-files:
+cabal-version:       >=1.10
+
+library
+  hs-source-dirs:      src
+  exposed-modules:     Lib
+  build-depends:       base
+  default-language:    Haskell2010
+
+executable symengine-hs-exe
+  hs-source-dirs:      app
+  main-is:             Main.hs
+  ghc-options:         -threaded -rtsopts -with-rtsopts=-N 
+  include-dirs:        /usr/local/include/symengine/
+  extra-libraries:     symengine stdc++ gmpxx gmp
+  build-depends:       base >= 4.7 && < 5
+                     , symengine-hs
+  default-language:    Haskell2010
+
+test-suite symengine-hs-test
+  type:                exitcode-stdio-1.0
+  hs-source-dirs:      test
+  main-is:             Spec.hs
+  build-depends:       base
+                     , symengine-hs
+  ghc-options:         -threaded -rtsopts -with-rtsopts=-N
+  include-dirs:        /usr/local/include/symengine/
+  extra-libraries:     symengine stdc++ gmpxx gmp
```

Comment:
The install location can be specified as a command line argument to the
build tool. The linking of libraries, I'll have to look into
On Tue, 8 Mar 2016 at 14:53 Isuru Fernando <notifications@github.com> wrote:

> In symengine-hs.cabal
> <https://github.com/symengine/symengine.hs/pull/1#discussion_r55333946>:
>
> > +  ghc-options:         -threaded -rtsopts -with-rtsopts=-N
> > +  include-dirs:        /usr/local/include/symengine/
> > +  extra-libraries:     symengine stdc++ gmpxx gmp
> > +  build-depends:       base >= 4.7 && < 5
> > +                     , symengine-hs
> > +  default-language:    Haskell2010
> > +
> > +test-suite symengine-hs-test
> > +  type:                exitcode-stdio-1.0
> > +  hs-source-dirs:      test
> > +  main-is:             Spec.hs
> > +  build-depends:       base
> > +                     , symengine-hs
> > +  ghc-options:         -threaded -rtsopts -with-rtsopts=-N
> > +  include-dirs:        /usr/local/include/symengine/
> > +  extra-libraries:     symengine stdc++ gmpxx gmp
>
> There are several optional libraries which need to be linked with if
> SymEngine is built with them. Also, SymEngine may be installed at a
> different location
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/symengine/symengine.hs/pull/1/files#r55333946>.
>